### PR TITLE
Make sure tests are cleaned up when they complete

### PR DESCRIPTION
**Pull request by rhatdan**
Fixes: https://github.com/containers/podman/issues/13789

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@vrothberg @cdoern PTAL

File: test/e2e/build_test.go [hunk 1]
```diff
@@ -786,17 +786,18 @@ RUN ls /dev/test1`, ALPINE)
...
+		err = os.Mkdir(buildRoot, 0755)
 		Expect(err).To(BeNil())
 		err = ioutil.WriteFile(containerFilePath, []byte(containerFile), 0755)
 		Expect(err).To(BeNil())
-		build := podmanTest.Podman([]string{"build", "-f", "./relative/Containerfile", "./relative/build-root"})
+		build := podmanTest.Podman([]string{"build", "-f", containerFilePath, buildRoot})
 		build.WaitWithDefaultTimeout()
 		Expect(build).To(Exit(0))
-		err = os.RemoveAll("relative")
-		Expect(err).To(BeNil())
```
Comment (TomSweeneyRedHat) [inline test/e2e/build_test.go:800]: shouldn't this be kept, or converted to use `relativeDir`?

**Comment by TomSweeneyRedHat** [comment]
Maybe it's end of day review brain, but I'm not sure how this is making sure the tests are cleaned upon completion via these changes?  I like the new variable definitions though.

**Comment by rhatdan** [comment]
@TomSweeneyRedHat Read the issue that is tied to it.  Basically if you create directories or files in TempDir they are removed. @vrothberg found that these files were not being removed.

File: test/e2e/build_test.go (hunk 1 above)
Comment (cdoern) [reply test/e2e/build_test.go:800]: I think this is ok to rm since he put all of the new files in `podmanTest.TempDir`

**Review by cdoern — APPROVED** [review]
LGTM, thanks for catching this

**Review by vrothberg — COMMENTED** [review]
/lgtm

Thanks!

**Pull request merged**